tests/v2/validation/scaling/custom_cluster_nodes.go

```diff
@@ -0,0 +1,86 @@
+package scaling
+
+import (
+	"github.com/rancher/norman/types"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/pkg/nodes"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	clusterNameLabel = "cluster.x-k8s.io/cluster-name"
+	workloadNS       = "default"
+	clusterNS        = "fleet-default"
+)
+
+func validateNodeCount(client *rancher.Client, clusterID string, nodeType string, initialNodeCount int) (bool, error) {
+	if nodeType == "worker" {
+		workerNodes, err := client.Management.Node.List(&types.ListOpts{
+			Filters: map[string]interface{}{
+				"clusterId": clusterID,
+				"worker":    true,
+			},
+		})
+		if err != nil {
+			return false, err
+		}
+		return len(workerNodes.Data) == initialNodeCount+1, err
```

Comment:
can you add some clarity on why there are hard coded ints here? Looking at the test itself, I think that should be a parameter. 

Probably the best options is to adapt what Markus has in the other PR, where this is using nodesAndRoles instead of manually entered strings/booleans. (see later comment)